pysmt/test/examples.py

```diff
@@ -632,6 +632,13 @@ def get_full_example_formulae(environment=None):
                     logic=pysmt.logics.get_logic_by_name("QF_ALIA*")
                 ),
 
+            Example(hr="((Array{Int, Int}(0)[1 := 3] = aii) & (aii[1] = 3))",
+                    expr=And(Equals(Array(INT, Int(0), {Int(1) : Int(3)}), aii), Equals(Select(aii, Int(1)), Int(3))),
+                    is_valid=False,
+                    is_sat=True,
+                    logic=pysmt.logics.get_logic_by_name("QF_ALIA*")
+                ),
+
```

Comment:
More tests are needed. In particular, cases in which : 
* Array Value does not have additional values
* Array Value has an even number of additional values (n > 3)
* Array Value has an odd number of additional Values (n > 3)

This are not necessarily Example's, but we can make them for regression.